### ISSUE TITLE
ansible-galaxy  init test-role creates 'init' directory, ignores role name

### ISSUE THREAD

**Issue by basictheprogram** [ansible/ansible#17563]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

ansible-galaxy
##### ANSIBLE VERSION

```
$ ansible --version
ansible 2.2.0 (devel c654a4617a) last updated 2016/09/14 10:05:44 (GMT -500)
  lib/ansible/modules/core: (detached HEAD ae6992bf8c) last updated 2016/09/13 10:06:39 (GMT -500)
  lib/ansible/modules/extras: (detached HEAD afd0b23836) last updated 2016/09/13 10:06:39 (GMT -500)
  config file =
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

Default configuration
##### OS / ENVIRONMENT

OS X 10.11.6
##### SUMMARY

ansible-galaxy init rolename creates a directory called init, ignores rolename
##### STEPS TO REPRODUCE

ansible-galaxy  init test-role
- init was created successfully

```
$ ansible-galaxy -vvvv init test-role
No config file found; using defaults
Opened /Users/tanner/.ansible_galaxy
Initial connection to galaxy_server: https://galaxy.ansible.com
Base API: https://galaxy.ansible.com/api/v1
https://galaxy.ansible.com/api/v1/platforms/?page_size
- init was created successfully
```
##### EXPECTED RESULTS

A role created in a directory called test-role
##### ACTUAL RESULTS

A role created in a directory called init

**Comment by basictheprogram** [comment]
Quick debug shows role_name getting set to init here:

```
ansible/lib/ansible/cli/galaxy.py(178)execute_init()
-> role_name = self.args.pop(0).strip() if self.args else None
(Epdb) self.args
['init', 'test-role']
```

Which I think is because init.py:set_action() is not working?

Don't really understand the logic of init.py:set_action()

**Comment by bcoca** [comment]
this is probably an effect of the recent changes to make it py3 compatible

**Comment by basictheprogram** [comment]
galaxy.py(123)parse() changes self.args.

Before the call self.args is ["ansible-galaxy", "test-role"] after the called to parse it's ['init', 'test-role']

```
0: self.args = [u'/Users...e-galaxy', u'test-role']

> /Users/tanner/projects/ansible.git/ansible/lib/ansible/cli/galaxy.py(123)parse()
-> self.options, self.args =self.parser.parse_args()
(Epdb)

0: self.args = ['init', 'test-role']
```

**Comment by basictheprogram** [comment]
Super hackery that fixes my problem.

```
diff --git a/lib/ansible/cli/galaxy.py b/lib/ansible/cli/galaxy.py
index 457e15d..db8cbed 100644
--- a/lib/ansible/cli/galaxy.py
+++ b/lib/ansible/cli/galaxy.py
@@ -175,7 +175,7 @@ class GalaxyCLI(CLI):
         force      = self.get_opt('force', False)
         offline    = self.get_opt('offline', False)

-        role_name = self.args.pop(0).strip() if self.args else None
+        role_name = self.args.pop(1).strip() if self.args else None
         if not role_name:
             raise AnsibleOptionsError("- no role name specified for init")
         role_path = os.path.join(init_path, role_name)
```

**Comment by abadger** [comment]
@basictheprogram  Cool, the fact that popping the first element  fixes this for you probably means that PR #17569 will fix it for you as well.

**Comment by abadger** [comment]
I've merged the fix in #17569 to devel.  Could you give it a test and close this issue if it's now fixed?
